### DISCUSSION TITLE
What is the idea behind train_on_inputs: false?

### DISCUSSION THREAD

**Discussion by jooray** [axolotl-ai-cloud/axolotl#890] [General]
Hello,

what is the idea behind "train on inputs" setting? At what point is the input masked when set to false? What does it achieve?

My question is if the model still retains ability to answer specific questions, just does not learn from formulations of the questions (which might be noisy if they come from real-world users), or if it is just not used for error propagation...

In what situations should we set this to false and when to true?

Thanks.

**Comment by dblakely** [comment]
The user's inputs are still fed into the model, so the model sees them while training. But they don't contribute to the language modeling loss -- loss is just for the output (response/answer) tokens. The intuition behind it is that you don't need to train the model to mimic the user inputs, you just need to train it to mimic the outputs _given_ the user inputs.
